elastic/kibana
# [Infra] Refactor routes context providers and inventory alert flyout

closes [144387](https://github.com/elastic/kibana/issues/144387)

## Summary

This PR changes the context Waffle context providers usage, so that they are only used in the Inventory UI page. Therefore, removes the `waffleTime`, `waffleOptions` and `waffleFilters` query parameters from other Infrastructure pages:

Hosts View route example:

**Before**
```
https://edge-oblt.kb.us-west2.gcp.elastic-cloud.com/app/metrics/hosts?
  waffleFilter=(expression:%27%27,kind:kuery)&
  waffleTime=(currentTime:1708347453400,isAutoReloading:!f)&
  waffleOptions=(accountId:%27%27,autoBounds:!t,boundsOverride:(max:1,min:0),customMetrics:!(),customOptions:!(),groupBy:!(),legend:(palette:cool,reverseColors:!f,steps:10),metric:(type:cpu),nodeType:host,region:%27%27,sort:(by:name,direction:desc),source:default,timelineOpen:!f,view:map)&
  _a=(dateRange:(from:now-15m,to:now),filters:!(),limit:500,panelFilters:!(),query:(language:kuery,query:%27%27))&
  controlPanels=(cloud.provider:(explicitInput:(fieldName:cloud.provider,id:cloud.provider,title:%27Cloud%20Provider%27),grow:!f,order:1,type:optionsListControl,width:medium),host.os.name:(explicitInput:(fieldName:host.os.name,id:host.os.name,title:%27Operating%20System%27),grow:!f,order:0,type:optionsListControl,width:medium),service.name:(explicitInput:(fieldName:service.name,id:service.name,title:%27Service%20Name%27),grow:!f,order:2,type:optionsListControl,width:medium))
```


**Now**
```
http://localhost:5601/ftw/app/metrics/hosts?
  _a=(dateRange:(from:now-15m,to:now),filters:!(),limit:100,panelFilters:!(),query:(language:kuery,query:%27%27))&
  controlPanels=(cloud.provider:(explicitInput:(fieldName:cloud.provider,id:cloud.provider,title:%27Cloud%20Provider%27),grow:!f,order:1,type:optionsListControl,width:medium),host.os.name:(explicitInput:(fieldName:host.os.name,id:host.os.name,title:%27Operating%20System%27),grow:!f,order:0,type:optionsListControl,width:medium),service.name:(explicitInput:(fieldName:service.name,id:service.name,title:%27Service%20Name%27),grow:!f,order:2,type:optionsListControl,width:medium))
```

**NOTE**: I had to refactor some alerting components because they were depending on the `WaffleOptions` context to retrieve some properties that are only relevant within the Inventory UI context

### How to test

- Start a local Kibana instance
- Navigate to the pages below and confirm that they work. `waffle` query variables only exist in the Inventory UI
  - Inventory
    - Navigate to Pod details 
    - Create and load Saved Views
    - Create Inventory and Metrics alerts
  - Metrics
    - Create and load Saved Views
    - Create Inventory and Metrics alerts
  - Hosts View
    - Navigate Hosts Details
